lib/http-client.ts

```diff
@@ -68,6 +114,92 @@ export class HttpClient {
         return this.send('DELETE', url, null, headers);
     }
 
+    /**
+     * 
+     * Method used to generate predefined reusable methods for given set of endpoints.
+     * 
+     * Example:
+     * 
+     * let endpoints =  { 
+     *      getPosts : {
+     *          path : "/posts",
+     *          "methods" : "GET"
+     *      }
+     *  }
+     * 
+     * Below method will create a dynamic function in HttpClient prototype with name "getPosts" 
+     * and the method can be directly called from test like `http.getPosts()`.
+     * 
+     * @param endpoints
+     * @returns {HttpClient}
+     */
+    
+    registerEndpoints(endpoints:IendPointTemplate):HttpClient {
+        let self:HttpClient = this;
+        Object.keys(endpoints).forEach(function (endPointName:string) {
+            var endPointDetails = endpoints[endPointName];
+            HttpClient.prototype[endPointName] = function (parameters?:any, body?:any, headers?:any):ResponsePromise {
+                //parse wildcard routes before making API call.
+                let parsedUrl = HttpClient.parseWildCardRoutes(endPointDetails.path, parameters || {});
+                return self.send(endPointDetails.method, parsedUrl, body, endPointDetails.headers || headers);
+            }
+        });
+        return this;
+    }
+
+    /**
+     * 
+     * Method to convert wildcard routes with their respective values.
+     * 
+     * Example 1:
+     * let path = "/products/{productId}";
+     * let wildcardParam = { productId : 1 };
+     * 
+     * OUTPUT: "/products/1"
+     * 
+     * Example 2:
+     * 
+     * If the path doesn't have any wildcard pattern, then this method will append parameter as url-encoded-form.
+     * 
+     * let path = "/products";
+     * let wildcardParam = { productId : 1 };
+     * 
+     * OUTPUT: "/products?productId=1"
+     * 
+     * @param path
+     * @param wildcardParam
+     * @returns {string}
+     */
+    private static parseWildCardRoutes(path:string, wildcardParam:any):string {
+        var regexObject,isFirstUrlParam = true;
+        for (var param in wildcardParam) {
+            if (wildcardParam.hasOwnProperty(param) && path.indexOf("{" + param + "}") >= 0) {
+                regexObject = new RegExp("{" + param + "}", "g");
+                path = path.replace(regexObject, HttpClient.convertParamsToStringObject(wildcardParam[param]));
+            } else {
+                if(isFirstUrlParam) {
+                    path += "?";
+                    isFirstUrlParam = false;
+                } 
+                // Check if end of url contains &.
+                path += (path.substr(path.length - 1) == "&" || path.substr(path.length - 1) == "?") ?
+                param + "=" + HttpClient.convertParamsToStringObject(wildcardParam[param]) :
```

Comment:
⚠️ what would happen if I put non-ASCII characters into the parameter value? the values in the URL should be URLEncoded. Also make sure that it's covered with tests (use e.g. chinese characters, emojis, ...)

Reply:
I think we can create a new feature request and tackle it separately as it requires considerable amount of time for testing with all possible usecases. Also we need to create additional endpoints in server side that accepts ASCII characters in URL params.

Reply:
ok, please, after I merge this PR, create an issue so that I won't forget about it